File: applications/credhub-api/src/test/java/org/cloudfoundry/credhub/TestPasswordKeyProxyFactory.java
package org.cloudfoundry.credhub;

import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;

import org.cloudfoundry.credhub.config.EncryptionKeyMetadata;
import org.cloudfoundry.credhub.services.InternalEncryptionService;
import org.cloudfoundry.credhub.services.KeyProxy;
import org.cloudfoundry.credhub.services.PasswordBasedKeyProxy;
import org.cloudfoundry.credhub.services.PasswordKeyProxyFactory;

@Service
@SuppressWarnings("unused")
@Profile("unit-test")
public class TestPasswordKeyProxyFactory implements PasswordKeyProxyFactory {
  @Override
  public KeyProxy createPasswordKeyProxy(
    final EncryptionKeyMetadata encryptionKeyMetadata, final InternalEncryptionService encryptionService) {
    return new PasswordBasedKeyProxy(encryptionKeyMetadata.getEncryptionPassword(), 1, encryptionService);
  }
}

File: backends/credhub/src/test/kotlin/org/cloudfoundry/credhub/utils/TestPasswordKeyProxyFactory.kt
package org.cloudfoundry.credhub.utils

import org.cloudfoundry.credhub.config.EncryptionKeyMetadata
import org.cloudfoundry.credhub.services.InternalEncryptionService
import org.cloudfoundry.credhub.services.KeyProxy
import org.cloudfoundry.credhub.services.PasswordBasedKeyProxy
import org.cloudfoundry.credhub.services.PasswordKeyProxyFactory
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Service

@Service
@Profile("unit-test")
class TestPasswordKeyProxyFactory : PasswordKeyProxyFactory {
    override fun createPasswordKeyProxy(
        encryptionKeyMetadata: EncryptionKeyMetadata,
        encryptionService: InternalEncryptionService
    ): KeyProxy {
        return PasswordBasedKeyProxy(encryptionKeyMetadata.encryptionPassword, 1, encryptionService)
    }
}
